cognite/powerops/config.py

```diff
@@ -818,6 +818,23 @@ def no_duplicated_scenarios(cls, value: dict[str, PriceScenario]):
             )
         return value
 
+    @root_validator(pre=False)
+    def rkom_process_ext_ids(cls, values):
+        for rkom_bid_config in values["rkom_bid_combination"]:
```

Comment:
A few things.
1. This is a validation of the `rkom_bid_combination` and not a combination of fields, so I would not use a `root_validator`, instead I would use validator on that fields. Note you can access other fields by using a values argument in the validator function. Furthermore, note that the values available depends on the order the fields appears in the class. So You will have to reorder based on which other fields you need in the validation. Also remember you can input `@validator(each=True)` to run validation on each element in a list.
2. I think the validation you are doing is good, but I also understood that the validation we asked for is a very specific schema `POWEROPS_<watercourse>_<auction>_<up/down>_<block>_<no. price scenarions (X-prices)>_<first volume>MW-<last volume>MW`. Thus I suggest using a regex to pick out the piece and validate each of them.

Reply:
A very rough sketch could look something like this
```python
    @validator("rkom_bid_combination", each_item=True)
    def valid_process_external_id(cls, value: RKOMBidCombinationConfig, values: dict):
        available_watercourses = {watercourse.name for watercourse in values["watercourses"]}

        for external_id in value.rkom_bid_config_external_ids:
            watercouse, auction, direction, = regex_external_id(external_id) 
            assert watercouse in available_watercourses
            assert auction == value.auction
            ...
        
        return value
```